### PR TITLE
Handle regex exception when comparing log

### PR DESCRIPTION
**Pull request by mazimkhan**
Lines from template log are first matched literally and then as regex. In failure cases a line that is expected to match literally will not match and be tried as a regex patter. If the line has unqualified regex it may raise exception. Hence regex exception should be handled as a possibility and match() function should return False (no match) on exception.

@bridadan @adbridge Please review.

### CODE REVIEW

**Comment by adbridge** [comment]
@mazimkhan merge commits onto PRs are frowned upon (certainly for mbed-os) code as it makes patching hard. A rebase and force push would have been better....

File: mbed_host_tests/host_tests_runner/host_test_default.py [hunk 1]
```diff
@@ -527,6 +528,10 @@ def match_log(self, line):
         if self.compare_log_idx < len(self.compare_log):
             regex = self.compare_log[self.compare_log_idx]
             # Either the line is matched as is or it is checked as a regular expression.
-            if regex in line or re.search(regex, line):
-                self.compare_log_idx += 1
+            try:
+                if regex in line or re.search(regex, line):
+                    self.compare_log_idx += 1
+            except error:
```
Comment (bridadan) [inline mbed_host_tests/host_tests_runner/host_test_default.py]: Any chance we can use the actual regex exception here? Is there one that you catch explicitly here? I don't think it's generally good practice to catch ALL exceptions and fail silently.

Comment (mazimkhan) [inline mbed_host_tests/host_tests_runner/host_test_default.py]: Please see line https://github.com/ARMmbed/htrun/pull/146/files#diff-abaa341b8a118c20a730035041013df4R25
This is actually specific regex exception. I think the name "error" causes confusion.

**Comment by mazimkhan** [comment]
@adbridge Done.

File: mbed_host_tests/host_tests_runner/host_test_default.py (hunk 1 above)
Comment (bridadan) [inline mbed_host_tests/host_tests_runner/host_test_default.py]: You're totally right, my bad! And I learned something new today :smile: But man yeah, what a blissfully ambiguous error name, good going regex library lol.

**Pull request merged**